cloudant-client/src/test/java/com/cloudant/tests/LoggingTest.java

```diff
@@ -0,0 +1,249 @@
+        assertMessage("GET .*/testdb request", 0);
+        assertMessage("GET .*/testdb response 200 OK", 1);
+
+        // Store the current log size
+        int logsize = handler.logEntries.size();
+
+        // Make a second request to a different URL and check that nothing else was logged
+        client.executeRequest(Http.GET(client.getBaseUri()));
+        assertEquals("There should have been no more log entries", logsize, handler.logEntries
+                .size());
+    }
+
+    private String methodFilterPropName = "com.cloudant.http.filter.method";
+    @Test
+    public void httpMethodFilterLogging() throws Exception {
+        setAndAssertLogProperty(methodFilterPropName, "GET");
+        logger = setupLogger(HttpConnection.class, Level.FINE);
+
+        // Make a GET request
+        client.executeRequest(Http.GET(client.getBaseUri()));
+
+        // Check there were two log messages one for request and one for response
+        assertEquals("There should be 2 log messages", 2, handler.logEntries.size());
+        // Check the messages were the ones we expected
+        assertMessage("GET .* request", 0);
+        assertMessage("GET .* response 200 OK", 1);
+
+        // Store the current log size
+        int logsize = handler.logEntries.size();
+
+        // Make a PUT request to a different URL and check that nothing else was logged
+        client.executeRequest(Http.PUT(client.getBaseUri(), "text/plain").setRequestBody(""));
+        assertEquals("There should have been no more log entries", logsize, handler.logEntries
+                .size());
+    }
+
+    @Test
+    public void httpMethodFilterLoggingList() throws Exception {
+        setAndAssertLogProperty(methodFilterPropName, "PUT,GET");
+        logger = setupLogger(HttpConnection.class, Level.FINE);
+
+        // Make a GET request
+        client.executeRequest(Http.GET(client.getBaseUri()));
+
+        // Check there were two log messages one for request and one for response
+        assertEquals("There should be 2 log messages", 2, handler.logEntries.size());
+        // Check the messages were the ones we expected
+        assertMessage("GET .* request", 0);
+        assertMessage("GET .* response 200 OK", 1);
+
+        // Make a PUT request to a different URL and check that new messages are logged
+        client.executeRequest(Http.PUT(client.getBaseUri(), "text/plain").setRequestBody(""));
+
+        assertEquals("There should now be 4 log messages", 4, handler.logEntries.size());
+        // Check the messages were the ones we expected
+        assertMessage("PUT .* request", 2);
+        assertMessage("PUT .* response 200 OK", 3);
+    }
+
+    @Test
+    public void clientBuilderLogging() throws Exception {
+        logger = setupLogger(ClientBuilder.class, Level.CONFIG);
+        CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer).build();
+        assertEquals("There should be 5 log entries", 5, handler.logEntries.size());
```

Comment:
Comment should explain what the 5 messages are so we can deal with regressions easily